### PR TITLE
retrieving old pause states and reasons

### PR DESCRIPTION
**Pull request by elena-vi**

### CODE REVIEW

**Comment by elena-vi** [comment]
[[Bug] Pause time stamp format not human in arrears action diary entry](https://trello.com/c/ciu3yWxa/51-bug-pause-time-stamp-format-not-human-in-arrears-action-diary-entry)

File: lib/hackney/income/tenancy_gateway.rb [hunk 1]
```diff
@@ -125,6 +125,22 @@ def update_tenancy(user_id:, tenancy_ref:, is_paused_until_date:, pause_reason:,
         res
       end
 
+      def get_tenancy_pause(tenancy_ref:)
+        uri = URI.parse(File.join(@api_host, "/tenancies/#{ERB::Util.url_encode(tenancy_ref)}/pause"))
+        req = Net::HTTP::Get.new(uri)
+        req['X-Api-Key'] = @api_key
+
+        res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) { |http| http.request(req) }
+
```
Comment (cob16) [inline lib/hackney/income/tenancy_gateway.rb]:
```suggestion
     unless res.is_a? Net::HTTPSuccess
          Exceptions::IncomeApiError.new(res), "when trying to get_tenancy_pause using '#{uri}'"
     end
```

File: lib/hackney/income/tenancy_gateway.rb [hunk 2]
```diff
@@ -125,6 +125,25 @@ def update_tenancy(user_id:, tenancy_ref:, is_paused_until_date:, pause_reason:,
...
 
+      def get_tenancy_pause(tenancy_ref:)
+        uri = URI.parse(File.join(@api_host, "/tenancies/#{ERB::Util.url_encode(tenancy_ref)}/pause"))
+        req = Net::HTTP::Get.new(uri)
+        req['X-Api-Key'] = @api_key
+
+        res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: false) { |http| http.request(req) }
+        # res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) { |http| http.request(req) }
+
+        raise Exceptions::IncomeApiError.new(res), "when trying to get_tenancy_pause using '#{uri}'" unless res.is_a? Net::HTTPSuccess
```
[the remark this answers is not in the document]
Comment (elena-vi) [reply lib/hackney/income/tenancy_gateway.rb]: this is logged out on the api side, do we need it in both apps?

File: lib/hackney/income/tenancy_gateway.rb [hunk 3]
```diff
@@ -125,6 +125,25 @@ def update_tenancy(user_id:, tenancy_ref:, is_paused_until_date:, pause_reason:,
         res
       end
 
+      def get_tenancy_pause(tenancy_ref:)
+        uri = URI.parse(File.join(@api_host, "/tenancies/#{ERB::Util.url_encode(tenancy_ref)}/pause"))
+        req = Net::HTTP::Get.new(uri)
+        req['X-Api-Key'] = @api_key
+
+        res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: false) { |http| http.request(req) }
+        # res = Net::HTTP.start(uri.hostname, uri.port, use_ssl: true) { |http| http.request(req) }
```
[the remark this answers is not in the document]
Comment (elena-vi) [reply lib/hackney/income/tenancy_gateway.rb]: yes

File: lib/hackney/income/tenancy_gateway.rb (hunk 2 above)
Comment (cob16) [inline lib/hackney/income/tenancy_gateway.rb]: we should throw a separate exception for 404 if we want to catch that one

**Pull request merged**